Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift

```diff
@@ -345,7 +345,12 @@ extension WKIntelligenceReplacementTextEffectCoordinator: PlatformIntelligenceTe
 
         if chunk.finished {
             self.replacementQueue.removeFirst()
-            await self.restoreSelectionAcceptedReplacements(true)
+            if let onFlushCompletion = self.onFlushCompletion {
+                await onFlushCompletion()
```

Comment:
I would add a comment above this line (above 349 and below 348) saying "At this point, the last chunk has been received, and the session has ended, so the flush completion handler can now be invoked now that the animation is fully complete. There is no need to additionally restore the selection here since the selection would have already been restored when the session ended."